mockserver.js

```diff
@@ -131,29 +147,21 @@ var mockserver = {
             });
         }
 
-
         // Now, permute the possible headers, and look for any matching files, prioritizing on
         // both # of headers and the original header order
         var content,
             permutations = [[]];
+
         if(headers.length) {
-            permutations = Combinatorics.permutationCombination(headers).toArray().sort(function(a, b) {return b.length - a.length});
+            permutations = Combinatorics.permutationCombination(headers).toArray().sort(function(a, b) { return b.length - a.length; });
             permutations.push([]);
         }
+
         while(permutations.length) {
-            var mockName =  method + permutations.pop().join('') + getBodyOrQueryString(body, query) + '.mock';
-            var mockFile = join(mockserver.directory, path, mockName);
-            
-            if(fs.existsSync(mockFile)) {
-                try {
-                    content = fs.readFileSync(mockFile, {encoding: 'utf8'});
-                } catch(err) {
-                    // ignore file read errors, maybe we matched something by accident
-                }
-            }
+            var prefix = method + permutations.pop().join('');
+            content = getMockedContent(path, prefix, body, query) || content;
```

Comment:
moved some stuff in a function since my original intention was to allow a lose match on key-values of choice, but as for now it would have been too complicated.
I'm not sure of the params' order tho.